### PR TITLE
stm32h573i_dk : Fix CAN core clock and add I2C test 

### PR DESCRIPTION
**Pull request by marwaiehm-st**
Set the PLL1_Q divider to 6 give a can core clock of 80MHz to resolve fdcan_clk reception problem because M_CAN requires that the host clock "APB1" should be higher or equal to the CAN core clock "PLL1_Q".
Add necessary overlay and stm32h573i_dk in i2c_target_api test case to enable the board.

### CODE REVIEW

**Review by gautierg-st — DISMISSED** [review]

File: tests/drivers/i2c/i2c_target_api/boards/stm32h573i_dk.overlay [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ * Short Pin PB7 to PB11, and PB6 to PB10, for the test to pass.
+ */
+
+&i2c1 {
+       clocks = <&rcc STM32_CLOCK_BUS_APB1 0x00200000>,
+		<&rcc STM32_SRC_PLL3_R I2C1_SEL(1)>;
+	pinctrl-0 = <&i2c1_scl_pb6 &i2c1_sda_pb7>;
+	pinctrl-names = "default";
+	clock-frequency = <I2C_BITRATE_FAST>;
+	status = "okay";
```
Comment (erwango) [inline tests/drivers/i2c/i2c_target_api/boards/stm32h573i_dk.overlay:22]: Not required as provided by board, unless there is a specific reason to duplicate it

File: tests/drivers/i2c/i2c_target_api/boards/stm32h573i_dk.overlay [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	};
+};
+
+&i2c2 {
+       clocks = <&rcc STM32_CLOCK_BUS_APB1 0x00400000>,
+		<&rcc STM32_SRC_PLL3_R I2C2_SEL(1)>;
+	pinctrl-0 = <&i2c2_scl_pb10 &i2c2_sda_pb11>;
+	pinctrl-names = "default";
+	clock-frequency = <I2C_BITRATE_FAST>;
+	status = "okay";
```
Comment (erwango) [inline tests/drivers/i2c/i2c_target_api/boards/stm32h573i_dk.overlay:36]: ditto

**Review by erwango — CHANGES_REQUESTED** [review]

File: tests/drivers/i2c/i2c_target_api/boards/stm32h573i_dk.overlay (hunk 1 above)
Comment (marwaiehm-st) [reply tests/drivers/i2c/i2c_target_api/boards/stm32h573i_dk.overlay:22]: Yes you are right there no specific reason to duplicate, i will remove it

File: tests/drivers/i2c/i2c_target_api/boards/stm32h573i_dk.overlay (hunk 2 above)
Comment (marwaiehm-st) [reply tests/drivers/i2c/i2c_target_api/boards/stm32h573i_dk.overlay:36]: I will remove it

**Review by Desvauxm-st — DISMISSED** [review]

**Review by erwango — DISMISSED** [review]

**Review by gautierg-st — DISMISSED** [review]

**Comment by marwaiehm-st** [comment]
Related to the issue: https://github.com/zephyrproject-rtos/zephyr/issues/70557

**Comment by erwango** [comment]
@marwaiehm-st Please update PR title

**Review by gautierg-st — APPROVED** [review]

File: boards/st/stm32h573i_dk/stm32h573i_dk.dts [hunk 3]
```diff
@@ -81,7 +81,7 @@
 	div-m = <5>;
 	mul-n = <96>;
 	div-p = <2>;
-	div-q = <2>;
+	div-q = <6>;
```
Comment (FRASTM) [inline boards/st/stm32h573i_dk/stm32h573i_dk.dts:84]: Did you check that changing the pll1_q output freq does not disturb other peripherals that have pll1_q as clock source

**Review by FRASTM — COMMENTED** [review]
else LGTM

File: boards/st/stm32h573i_dk/stm32h573i_dk.dts (hunk 3 above)
Comment (marwaiehm-st) [reply boards/st/stm32h573i_dk/stm32h573i_dk.dts:84]: Yes, i checked it. The pll1_q  output frep is only used by the peripheral FDCAN  on stm32h573i_dk.

**Review by erwango — APPROVED** [review]

**Review by FRASTM — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]

**Comment by henrikbrixandersen** [comment]
With this change in, you should be able to add `CONFIG_TEST_ALL_BITRATES=y` for this board to `tests/drivers/can/timing/`.

**Pull request merged**